library/Businessprocess/Monitoring.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Icinga\Module\Businessprocess;
+
+use Icinga\Application\Icinga;
+use Icinga\Module\Monitoring\Controller;
+use Icinga\Module\Monitoring\DataView\DataView;
+
+class Monitoring extends Controller
+{
+    /**
+     * Cache for {@link getInstance()}
+     *
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * Factory
+     *
+     * @return self
+     */
+    final private static function getInstance()
+    {
+        if (static::$instance === null) {
+            static::$instance = new self(Icinga::app()->getRequest(), Icinga::app()->getResponse());
```

Comment:
This seems like a pretty ugly hack

Reply:
I can't use arbitrary info from the monitoring backend.
I have to respect the current user's restrictions – with the code provided by the current IW2 release.

Reply:
I just wished there would be a better way... *sigh* :wink: